docs/advanced-topics/best-practices/best-practices-secure-fair-workflows.rst

```diff
@@ -0,0 +1,174 @@
+
+- Wrap your pipeline in one or more workflow languages supported by Dockstore:
+	- `Common Workflow Language (CWL) <https://docs.dockstore.org/en/develop/getting-started/getting-started-with-cwl.html>`_
+    	- Used by SevenBridges (BioData Catalyst, Cancer Genomics Cloud)
+	- `Workflow Description Language (WDL) <https://docs.dockstore.org/en/develop/getting-started/getting-started-with-wdl.html>`_
+		- Used by Terra (BioData Catalyst, AnVIL), DNAnexus
+	- `Galaxy <https://docs.dockstore.org/en/develop/getting-started/getting-started-with-galaxy.html?highlight=launch%20with%20biodata>`_
+		- Used by Terra (AnVIL)
+	- `NextFlow <https://docs.dockstore.org/en/develop/getting-started/getting-started-with-nextflow.html>`_
+- Provide a parameter file (JSON or YAML) containing example parameters used for launching your workflow. 
+	- The parameter file is where you should link to open access test data for your tool or workflow (learn more in :ref:`Reusable`).
+	- You can submit multiple parameter files so consider sharing one for a local run (you can use the `Dockstore Command Line Interface (CLI) <https://docs.dockstore.org/en/develop/launch-with/launch.html?highlight=json>`_ to launch tools and workflows locally) as well as examples for a launch-with partner (such as `BioData Catalyst <https://bdcatalyst.gitbook.io/biodata-catalyst-documentation/analyze-data/dockstore/launch-workflows-with-biodata-catalyst>`_ or `AnVIL <https://docs.dockstore.org/en/develop/launch-with/anvil-launch-with.html>`_).
+- Provide a checker tool/workflow by following this `guide <https://docs.dockstore.org/en/develop/advanced-topics/checker-workflows.html>`_. 
+	- Checker workflows are additional workflows you can associate with a tool or workflow. The purpose of them is to ensure that a tool or workflow, given some inputs, produces the expected outputs on a platform different from the one where you are developing.
+	- Providing a checker workflow gives other researchers confidence that they can run the work on their system correctly. 
+
+Reusable
+********
+
+- Best practices when referencing the image from the image repository is to provide the digest format of the image as an immutable record in the tool or workflow. Here is an example of a digest format:
+        
+
+::
+
+	task digestDocker {
+   		command {
+   			echo "hello world"
+   		 }
+    	runtime { 		docker:"pkrusche/hap.py@sha256:f63e020c4062e0be8d081a50de16562f2ba161166e896655868efdb5527a8640
+    		}	
+	}
+
+ 
+- Do not use parameterized images, as these formats change and cause the workflow to no longer be reproducible. Here is an example of what not to do: 
+
+::
+
+	task paramterizedDocker {
+		input {
+			String docker_image
+   		}
+   		command {
+   			echo "hello world"
+   		}
+    	runtime {
+    		doker: docker_image
+       	 	}	
+	}
+ 
+ 
+- Do not use untagged, “latest” tag, or a version tag, e.g. “v1”, as these formats change and cause the workflow to no longer be reproducible.
```

Comment:
I think you can separate out "v1" as a different case. I probably should have commented on your actual doc; missed the follow up.

v1 typically wouldn't change, but it could... People might be confused by this.